src/hub/views.py

```diff
@@ -504,6 +508,89 @@ def _get_latest_actions_context(self):
         }
         return context
 
+    @action(
+        detail=False,
+        methods=[GET],
+        permission_classes=[AllowAny]
+    )
+    def by_contributions(self, request, pk=None):
+        hub_id = request.GET.get('hub_id', None)
+        hub_qs = Hub.objects.all().distinct() if hub_id is None \
+            else Hub.objects.filter(id=hub_id)
+
+        timeframe_query = Q(
+            **resolve_timeframe_for_contribution(
+                request.GET.get('start_date', None),
+                request.GET.get('end_date', None),
+                'related_documents__contributions__created_date'
+            ),
+        )
+
+        total_contrib_query = Q(
+            related_documents__contributions__contribution_type__in=[
+                Contribution.COMMENTER,
+                Contribution.SUBMITTER,
+                Contribution.SUPPORTER,
+            ],
+        ) & timeframe_query
+
+        qs_key = 'related_documents__contributions__contribution_type'
+        comment_query = Q(
+            **dict([(qs_key, Contribution.COMMENTER)])
+        ) & timeframe_query
+
+        submission_query = Q(
+            **dict([(qs_key, Contribution.SUBMITTER)])
+        ) & timeframe_query
+
+        support_query = Q(
+            **dict([(qs_key, Contribution.SUPPORTER)])
+        ) & timeframe_query
+
+        order_by = '-total_contribution_count' \
+            if request.GET.get('order_by', 'desc') == 'desc' \
+            else 'total_contribution_count'
+
+        hub_qs_ranked_by_contribution = \
+            hub_qs.prefetch_related(
+                'related_documents',
+                'related_documents__contributions',
+                'related_documents__contributions__created_date__gte',
+            ).annotate(
+                total_contribution_count=Count(
+                    'id', filter=total_contrib_query
+                ),
+                comment_count=Count(
+                    'id', filter=comment_query
+                ),
+                submission_count=Count(
+                    'id', filter=submission_query
+                ),
+                support_count=Count(
+                    'id', filter=support_query
+                ),
+            ).order_by(order_by)
+
+        paginator = Paginator(
+            hub_qs_ranked_by_contribution,  # qs
+            10,  # page size
+        )
+        curr_page_number = request.GET.get('page') or 1
+        curr_pagation = paginator.page(curr_page_number)
+
+        return Response(
+            {
+                'count': paginator.count,
+                'has_more': curr_pagation.has_next(),
+                'page': curr_page_number,
+                'result': HubContributionSerializer(
+                    curr_pagation.object_list,
+                    many=True,
+                ).data,
+            },
+            status=200
+        )
+
```

Comment:
You need a manual call of pagination here because Hub already has an existing default pagination. This is written on top of a Django Viewset. But we have a special query set for this particular endpoint